src/notte/controller/base.py

```diff
@@ -94,18 +99,43 @@ async def execute_interaction_action(self, action: InteractionAction) -> Browser
         if action.press_enter is not None:
             press_enter = action.press_enter
         # locate element (possibly in iframe)
-        locator = await locale_element(self.window.page, action.selector)
+        locator: Locator = await locale_element(self.window.page, action.selector)
         original_url = self.window.page.url
 
         action_timeout = self.window.config.wait.action_timeout
 
+        def is_code(text: str) -> bool:
+            try:
+                lexer = guess_lexer(text)
+                return str(lexer) != "<pygments.lexers.TextLexer>"
+            except Exception:
+                return False
+
         match action:
             # Interaction actions
             case ClickAction():
                 await locator.click(timeout=action_timeout)
             case FillAction(value=value):
-                await locator.fill(value, timeout=action_timeout)
-                await self.window.page.wait_for_timeout(500)
+                if is_code(text=value):
+                    await locator.focus()
+                    selectallKey: Literal["Meta+A", "Control+A"] = (
+                        "Meta+A" if platform.system() == "Darwin" else "Control+A"
+                    )
+                    await self.window.page.keyboard.press(key=selectallKey)
+                    await self.window.page.keyboard.press(key="Backspace")
+
+                    await self.window.page.evaluate("text => {navigator.clipboard.writeText(text)}", value)
+
+                    # Paste using keyboard shortcut
+                    pasteKey: Literal["Meta+V", "Control+V"] = (
+                        "Meta+V" if platform.system() == "Darwin" else "Control+V"
```

Comment:
create a function `platform_control_key() -> Litteral["Darwin" | "Control"]` and remove all variables  `pasteKey', `copykey`, etc => to put directly `...press(key=f"{platform_control_key()+V"}` , etc.